contrib/TestHarness2/README.md

```diff
@@ -158,4 +97,69 @@ Key XML attributes:
 When TestHarness2 encounters fatal errors, it outputs an error XML document to stdout and exits with code 1. The error XML includes:
 - Error message and type
 - Joshua seed (if available)
-- Test UID (if available)
\ No newline at end of file
+- Test UID (if available)
+
+### Determinism Check Log Preservation
+
+TestHarness2 automatically preserves trace logs from both the initial run and determinism check when determinism verification occurs (approximately 5% of simulation tests). This feature prevents the loss of valuable debugging information that would otherwise be overwritten.
+
+**How it works:**
+1. **Initial run** completes and produces trace files
+2. **Files moved** to `trace_initial_run/` directory before determinism check
+3. **Determinism check** runs and creates new trace files in main directory
+4. **Determinism check files moved** to `trace_determinism_check/` directory after completion
+
+**Result:** Both sets of logs are clearly separated for easy analysis.
+
+**Example directory structure after determinism check:**
+```
+<test_uid>/
+├── trace_initial_run/
+│   └── trace.0.0.0.0.267.1752628294.yb4TCb.0.1.json    # Initial run
+├── trace_determinism_check/
+│   └── trace.0.0.0.0.505.1752628352.yb4TCb.0.1.json    # Determinism check run
+├── simfdb/
+└── valgrind-3866502111.xml
+```
+
+## Joshua LogTool Integration
+
+TestHarness2 integrates with [`joshua_logtool.py`](../joshua_logtool.py) to automatically upload trace logs to a FoundationDB cluster for long-term storage and analysis when test failures occur.
```

Comment:
Maybe a TODO item: move `joshua_logtool.py` to be part of `TestHarness2`.

Reply:
nod